nncf/sparsity/rb/loss.py

```diff
@@ -67,3 +67,38 @@ def target_sparsity_rate(self):
 
     def statistics(self):
         return {'mean_sparse_prob': 1 - self.mean_sparse_prob}
+
+    def set_target_sparsity_loss(self, sparsity_level):
+        self.target = 1 - sparsity_level
+
+class PerLayerSparseLoss(SparseLoss):
+    def __init__(self, sparse_layers=None, target=1.0, p=0.05):
+        super().__init__(sparse_layers, target, p)
+        self.per_layer_target = {}
+        for sparse_layer in self._sparse_layers:
+            self.per_layer_target[sparse_layer] = self.target
+
+    def forward(self):
+        if self.disabled:
+            return 0
+
+        params = 0
+        loss = 0
+        sparse_prob_sum = 0
+        sparse_layers_loss = 0
+        for sparse_layer in self._sparse_layers:
+            if not self.disabled and not sparse_layer.sparsify:
+                raise AssertionError(
+                    "Invalid state of SparseLoss and SparsifiedWeight: mask is frozen for enabled loss")
+            if sparse_layer.sparsify:
+                sw_loss = sparse_layer.loss()
+                params_layer = sw_loss.view(-1).size(0)
+                params += params_layer
+                sparse_layers_loss += sw_loss.sum() / params_layer - self.per_layer_target[sparse_layer]
```

Comment:
It looks like that the loss does not apply a penalty for the following case `sw_loss.sum() / params_layer` > `self.per_layer_target[sparse_layer]`

Reply:
Done.